kbase-extension/static/kbase/js/common/cellComponents/filePathWidget.js

```diff
@@ -313,129 +334,136 @@ define([
-                    //             })
-                    //         ])
-                    //     );
-                    });
+            }).map(function (id) {
+                return params.specs[id];
             });
         }
 
+        function createParamWidget(appSpec, filePathParams, parameterId) {
+            const spec = filePathParams.paramMap[parameterId];
+            try {
+                return makeFieldWidget(appSpec, spec, initialParams[spec.id], filePathParams.layout)
+                    .then((widget) => {
+                        widgets.push(widget);
+                        return widget.start({
+                            node: document.getElementById(filePathParams.view[spec.id].id)
+                        });
+                    });
+            } catch (ex) {
+                console.error('Error making input field widget', ex);
+                const errorDisplay = div({
+                    class: 'kb-field-widget__error_message--file-paths'
+                }, [
+                    ex.message
+                ]);
+                document.getElementById(filePathParams.view[spec.id].id).innerHTML = errorDisplay;
+            }
+        }
+
+        function deleteRow(e){
+            $(e.target).closest('tr').remove();
+        }
+
+
+        function renderParameterRow(parameterRow, rowNumber) {
+            const appSpec = model.getItem('appSpec');
+
+            const params = model.getItem('parameters');
+            let filePathParams = makeParamsLayout(findPathParams(params));
+
+            if (!filePathParams.layout.length) {
+                ui.getElement(`${cssClassType}s-area`).classList.add('hidden');
+            } else {
+                parameterRow.innerHTML = div({
+                    class: `${cssBaseClass}__param_container row`,
+                }, [
+                    filePathParams.content
+                ]);
+
+                $(parameterRow).prepend(
+                    td({
+                        class: `${cssBaseClass}__file_number`,
+                    }, [
+                        rowNumber
+                    ])
+                );
+
+                $(parameterRow).append(
+                    td({}, [
+                        button({
+                            class: 'btn btn__text',
+                            type: 'button',
+                            id: events.addEvent({
+                                type: 'click',
+                                handler: function(e){
+                                    deleteRow(e);
+                                }
+
+                            })
+                        },[
+                            icon({
+                                class: 'fa fa-trash-o fa-lg',
+                            })
+                        ])
+                    ])
+                );
+
+                filePathParams.layout.map((parameterId) => {
+                    createParamWidget(appSpec, filePathParams, parameterId);
```

Comment:
This chunk of code use to be wrapped in a promise. I don't think it needs to be. Thoughts?

Reply:
agreed - seems like we don't need yet another promise call here

Reply:
Noted, I will put the promise in there.